src/device/pf_cmdev.c

```diff
@@ -353,7 +353,7 @@ void pf_cmdev_free_diag(
    }
    else
    {
-      printf("ix =%u\n", (unsigned)item_ix);
+      LOG_ERROR(PNET_LOG, "CMDEV(%d): Failed to free diag with index %u\n", __LINE__, (unsigned)item_ix);
```

Comment:
The error is not propagated here. Would an assert be more appropriate?

Reply:
You are right, this is a void xxx_free() function. An out-of-range index should be detected elsewhere (if it is a problem). Here is too late. I will remove the print-out.